### PR TITLE
[tests-only][full-ci] refactored the test and added skipOnStable 3.0 tag on deprovisioning notification test

### PR DESCRIPTION
**Pull request by ghost**
## Description
refactoring deprovisioning notification test 

## Related Issue

- Fixes https://github.com/owncloud/ocis/pull/7065#discussion_r1298317796

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Review by phil-davis — APPROVED** [review]

**Review by SagarGi — COMMENTED** [review]
LGTM :+1:

File: tests/acceptance/features/apiNotification/deprovisioningNotification.feature [hunk 1]
```diff
@@ -11,7 +12,6 @@ Feature: Deprovisioning notification
 
   Scenario: administrator creates a deprovisioning notification about shutting down the instance
     When the administrator creates a deprovisioning notification
-    Then the HTTP status code should be "200"
     When user "Alice" lists all notifications
```
Comment (phil-davis) [inline tests/acceptance/features/apiNotification/deprovisioningNotification.feature:15]: We end up having 2 `When`
```suggestion
    And user "Alice" lists all notifications
```

File: tests/acceptance/features/apiNotification/deprovisioningNotification.feature [hunk 2]
```diff
@@ -173,7 +173,6 @@ Feature: Deprovisioning notification
 
   Scenario Outline: administrator creates a deprovisioning notification with different date formats
     When the administrator creates a deprovisioning notification for date "<deprovision_date>" of format "<deprovision_date_format>"
-    Then the HTTP status code should be "200"
     When user "Alice" lists all notifications
```
Comment (phil-davis) [inline tests/acceptance/features/apiNotification/deprovisioningNotification.feature:176]:
```suggestion
    And user "Alice" lists all notifications
```

File: tests/acceptance/features/apiNotification/deprovisioningNotification.feature [hunk 3]
```diff
@@ -208,7 +207,6 @@ Feature: Deprovisioning notification
   Scenario Outline: administrator change a deprovisioning notification with different date formats
     Given the administrator has created a deprovisioning notification
     When the administrator creates a deprovisioning notification for date "<deprovision_date>" of format "<deprovision_date_format>"
-    Then the HTTP status code should be "200"
     When user "Alice" lists all notifications
```
Comment (phil-davis) [inline tests/acceptance/features/apiNotification/deprovisioningNotification.feature:210]:
```suggestion
    And user "Alice" lists all notifications
```

**Review by phil-davis — COMMENTED** [review]
What to do about the suggested small changes?
I suppose that we make them here, and backport to stable-3.0

**Review by phil-davis — APPROVED** [review]

File: tests/acceptance/features/apiNotification/deprovisioningNotification.feature [hunk 4]
```diff
@@ -11,8 +12,7 @@ Feature: Deprovisioning notification
 
   Scenario: administrator creates a deprovisioning notification about shutting down the instance
     When the administrator creates a deprovisioning notification
-    Then the HTTP status code should be "200"
-    When user "Alice" lists all notifications
+    And user "Alice" lists all notifications
     Then the HTTP status code should be "200"
```
Comment (saw-jan) [inline tests/acceptance/features/apiNotification/deprovisioningNotification.feature:16]:
```suggestion
    Then the HTTP status code of responses on each endpoint should be "200, 200" respectively
```

**Review by saw-jan — CHANGES_REQUESTED** [review]
We want to check status codes for both When steps

**Review by saw-jan — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]

File: tests/acceptance/features/bootstrap/SettingsContext.php [hunk 5]
```diff
@@ -315,7 +315,8 @@ public function sendRequestGetSettingsValuesList(string $user): void {
 			$this->spacesContext->sendPostRequestToUrl($fullUrl, $user, $this->featureContext->getPasswordForUser($user), $body)
 		);
 
-		$this->featureContext->theHTTPStatusCodeShouldBe(
+		Assert::assertEquals(
+			$this->featureContext->getResponse()->getStatusCode(),
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SettingsContext.php:319]: why was it required? I think `theHTTPStatusCodeShouldBe` does the same

**Review by saw-jan — CHANGES_REQUESTED** [review]
?

**Review by saw-jan — APPROVED** [review]

**Review by SagarGi — APPROVED** [review]
LGTM :+1:

File: tests/acceptance/features/bootstrap/NotificationContext.php [hunk 6]
```diff
@@ -549,6 +548,22 @@ public function userCreatesDeprovisioningNotification(
 		$this->featureContext->setResponse($response);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/NotificationContext.php:548]: can we move it to `theAdministratorCreatesADeprovisioningNotification`? do not set responses in helper functions
```suggestion
```

**Review by saw-jan — CHANGES_REQUESTED** [review]
small request

**Review by saw-jan — APPROVED** [review]

**Pull request merged**